databroker/experimental/server_ext.py

```diff
@@ -339,7 +335,13 @@ def new_variation(
             **kwargs,
         )
 
-    def post_metadata(self, metadata, structure_family, structure, specs, mimetype):
+    def post_metadata(self, metadata, structure_family, structure, specs):
+
+        mime_structure_association = {
```

Comment:
I think this could be at global scope. No reason to re-define it on every call of `post_metadata`.